src/cmp-ui.ts

```diff
@@ -1,25 +1,55 @@
-import { readGuCookie, readIabCookie } from './cookies';
-import { CMP_DOMAIN, CMP_READY_MSG, CMP_CLOSE_MSG } from './config';
+import { readIabCookie } from './cookies';
+import { logConsent } from './consent-logs';
+import {
+    CMP_DOMAIN,
+    CMP_READY_MSG,
+    CMP_CLOSE_MSG,
+    CMP_SAVED_MSG,
+} from './config';
 
-type Callback = () => void;
+type Callback = (error?: Error) => void;
 
 export const setupMessageHandlers = (
     onReadyCmp: Callback,
     onCloseCmp: Callback,
+    onErrorCmp: Callback,
 ): void => {
     const receiveMessage = (event: MessageEvent): void => {
+        const withError = (callback: Callback): void => {
+            try {
+                callback();
+            } catch (e) {
+                onErrorCmp(e);
+            }
+        };
+
         const { origin, data } = event;
 
         if (origin !== CMP_DOMAIN) {
             return;
         }
 
-        switch (data) {
+        const { msgType, msgData } = data;
+
+        switch (msgType) {
             case CMP_READY_MSG:
-                onReadyCmp();
+                withError(onReadyCmp);
                 break;
             case CMP_CLOSE_MSG:
-                onCloseCmp();
+                withError(onCloseCmp);
+                break;
+            case CMP_SAVED_MSG:
+                logConsent(msgData)
+                    .then(response => {
+                        if (!response.ok) {
+                            throw new Error(
```

Comment:
Is this error handled (e.g. logged) somewhere else? Be good to have visibility of these if possible.

Reply:
This error will be caught by the `catch` block which will pass it onto `onErrorCmp`, this function is passed into the library from `frontend` and logs these errors in Sentry.